packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/CameraAndroidCameraxPlugin.java

```diff
@@ -4,39 +4,62 @@
 
 package io.flutter.plugins.camerax;
 
+import android.content.Context;
 import androidx.annotation.NonNull;
 import io.flutter.embedding.engine.plugins.FlutterPlugin;
-import io.flutter.plugin.common.MethodCall;
-import io.flutter.plugin.common.MethodChannel;
-import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
-import io.flutter.plugin.common.MethodChannel.Result;
-
-/** CameraAndroidCameraxPlugin */
-public class CameraAndroidCameraxPlugin implements FlutterPlugin, MethodCallHandler {
-  /// The MethodChannel that will the communication between Flutter and native Android
-  ///
-  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
-  /// when the Flutter Engine is detached from the Activity
-  private MethodChannel channel;
+import io.flutter.embedding.engine.plugins.activity.ActivityAware;
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
+import io.flutter.plugin.common.BinaryMessenger;
+
+/** Platform implementation of the camera_plugin implemented with the CameraX library. */
+public final class CameraAndroidCameraxPlugin implements FlutterPlugin, ActivityAware {
+  private InstanceManager instanceManager;
+  private FlutterPluginBinding pluginBinding;
+
+  /**
+   * Initialize this within the {@code #configureFlutterEngine} of a Flutter activity or fragment.
+   *
+   * <p>See {@code io.flutter.plugins.camera.MainActivity} for an example.
+   */
+  public CameraAndroidCameraxPlugin() {}
+
+  void setUp(BinaryMessenger binaryMessenger, Context context) {
```

Comment:
`context` is not currently being used I noticed but I will need it in a follow up PR as per: https://github.com/flutter/plugins/pull/6307/files#diff-f037f7d2d2a92c6e1b775e086f9da043c0a77dfbb1209c63adf41b0b70899ed2R27-R40